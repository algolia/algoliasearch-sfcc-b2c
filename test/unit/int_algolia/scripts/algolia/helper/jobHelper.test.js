const MasterProductMock = require("../../../../../mocks/dw/catalog/MasterProduct");
const VariantMock = require("../../../../../mocks/dw/catalog/Variant");
const collectionHelper = require("../../../../../mocks/helpers/collectionHelper");
const jobHelper = require("../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/helper/jobHelper");

test('generateVariantRecordsWithColorVariations', () => {
    // master product with two size variations on the same color variation
    const masterProduct = new MasterProductMock();
    const variantPinkSize4 = new VariantMock({
        ID: '701644031206M',
        variationAttributes: { color: 'JJB52A0', size: '004' },
        masterProduct,
    });
    const variantPinkSize6 = new VariantMock({
        ID: '701644031213M',
        variationAttributes: { color: 'JJB52A0', size: '006' },
        masterProduct,
    });
    masterProduct.variants = collectionHelper.createCollection([
        variantPinkSize4,
        variantPinkSize6,
    ]);

    const variantRecords = jobHelper.generateVariantRecordsWithColorVariations({
        masterProduct,
        locales: collectionHelper.createCollection(['fr']),
        attributeList: ['name', 'primary_category_id', 'categories', 'in_stock', 'price', 'url'],
        nonLocalizedAttributes: [],
    });
    expect(variantRecords).toMatchSnapshot();
});

test('generateMasterRecords', () => {
    // master product with two size variations on the same color variation
    const masterProduct = new MasterProductMock();
    const variantPinkSize4 = new VariantMock({
        ID: '701644031206M',
        variationAttributes: { color: 'JJB52A0', size: '004' },
        masterProduct,
    });
    const variantPinkSize6 = new VariantMock({
        ID: '701644031213M',
        variationAttributes: { color: 'JJB52A0', size: '006' },
        masterProduct,
    });
    masterProduct.variants = collectionHelper.createCollection([
        variantPinkSize4,
        variantPinkSize6,
    ]);

    const variantRecords = jobHelper.generateMasterRecords({
        product: masterProduct,
        locales: collectionHelper.createCollection(['en']),
        masterAttributes: ['name', 'primary_category_id', 'categories'],
        variantAttributes: ['in_stock', 'price', 'url'],
        nonLocalizedAttributes: [],
    });
    expect(variantRecords).toMatchSnapshot();
});
