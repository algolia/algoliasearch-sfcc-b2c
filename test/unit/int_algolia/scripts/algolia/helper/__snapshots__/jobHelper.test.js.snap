// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateMasterRecords 1`] = `
{
  "en": algoliaLocalizedProduct {
    "__primary_category": {
      "0": "Womens",
      "1": "Womens > Clothing",
      "2": "Womens > Clothing > Bottoms",
    },
    "categories": [
      [
        {
          "id": "newarrivals-womens",
          "name": "Womens",
        },
      ],
      [
        {
          "id": "womens-clothing-bottoms",
          "name": "Bottoms",
        },
        {
          "id": "womens-clothing",
          "name": "Clothing",
        },
        {
          "id": "womens",
          "name": "Womens",
        },
      ],
    ],
    "color_variations": [
      {
        "color": "Hot Pink Combo",
        "image_groups": [
          {
            "_type": "image_group",
            "images": [
              {
                "_type": "image",
                "alt": "Floral Dress, Hot Pink Combo, large",
                "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dwcc434d54/images/large/PG.10237222.JJB52A0.PZ.jpg",
                "title": "Floral Dress, Hot Pink Combo",
              },
              {
                "_type": "image",
                "alt": "Floral Dress, Hot Pink Combo, large",
                "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw58a034a4/images/large/PG.10237222.JJB52A0.BZ.jpg",
                "title": "Floral Dress, Hot Pink Combo",
              },
            ],
            "view_type": "large",
          },
          {
            "_type": "image_group",
            "images": [
              {
                "_type": "image",
                "alt": "Floral Dress, Hot Pink Combo, small",
                "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw4e4ce4f6/images/small/PG.10237222.JJB52A0.PZ.jpg",
                "title": "Floral Dress, Hot Pink Combo",
              },
              {
                "_type": "image",
                "alt": "Floral Dress, Hot Pink Combo, small",
                "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw2612fb5e/images/small/PG.10237222.JJB52A0.BZ.jpg",
                "title": "Floral Dress, Hot Pink Combo",
              },
            ],
            "view_type": "small",
          },
          {
            "_type": "image_group",
            "images": [
              {
                "_type": "image",
                "alt": "Floral Dress, Hot Pink Combo, swatch",
                "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw506aa734/images/swatch/PG.10237222.JJB52A0.CP.jpg",
                "title": "Floral Dress, Hot Pink Combo",
              },
            ],
            "view_type": "swatch",
          },
        ],
        "variant_url": "/on/demandware.store/Sites-Algolia_SFRA-Site/en/Product-Show?pid=25592581M",
      },
    ],
    "default_variant_id": "701644031206M",
    "in_stock": true,
    "maxPrice": {
      "EUR": 92.88,
      "USD": 129,
    },
    "minPrice": {
      "EUR": 92.88,
      "USD": 129,
    },
    "name": "Floral Dress",
    "objectID": "25592581M",
    "price": {
      "EUR": 92.88,
      "USD": 129,
    },
    "primary_category_id": "womens-clothing-bottoms",
    "variants": [
      algoliaLocalizedProduct {
        "in_stock": true,
        "price": {
          "EUR": 92.88,
          "USD": 129,
        },
        "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/en/Product-Show?pid=701644031206M",
        "variantID": "701644031206M",
      },
      algoliaLocalizedProduct {
        "in_stock": true,
        "price": {
          "EUR": 92.88,
          "USD": 129,
        },
        "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/en/Product-Show?pid=701644031213M",
        "variantID": "701644031213M",
      },
    ],
  },
}
`;

exports[`generateVariantRecordsWithColorVariations 1`] = `
{
  "fr": [
    algoliaLocalizedProduct {
      "__primary_category": {
        "0": "Femmes",
        "1": "Femmes > Vêtements",
        "2": "Femmes > Vêtements > Bas",
      },
      "categories": [
        [
          {
            "id": "newarrivals-womens",
            "name": "Femmes",
          },
        ],
        [
          {
            "id": "womens-clothing-bottoms",
            "name": "Bas",
          },
          {
            "id": "womens-clothing",
            "name": "Vêtements",
          },
          {
            "id": "womens",
            "name": "Femmes",
          },
        ],
      ],
      "color_variations": [
        {
          "color": "Combo rose vif",
          "image_groups": [
            {
              "_type": "image_group",
              "images": [
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, large",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dwcc434d54/images/large/PG.10237222.JJB52A0.PZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, large",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw58a034a4/images/large/PG.10237222.JJB52A0.BZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
              ],
              "view_type": "large",
            },
            {
              "_type": "image_group",
              "images": [
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, small",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw4e4ce4f6/images/small/PG.10237222.JJB52A0.PZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, small",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw2612fb5e/images/small/PG.10237222.JJB52A0.BZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
              ],
              "view_type": "small",
            },
            {
              "_type": "image_group",
              "images": [
                {
                  "_type": "image",
                  "alt": "Robe fleurie, Mix rose vif, swatch",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw506aa734/images/swatch/PG.10237222.JJB52A0.CP.jpg",
                  "title": "Robe fleurie, Mix rose vif",
                },
              ],
              "view_type": "swatch",
            },
          ],
          "variant_url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Product-Show?pid=25592581M",
        },
      ],
      "in_stock": true,
      "name": "Robe florale",
      "objectID": "701644031206M",
      "price": {
        "EUR": 92.88,
        "USD": 129,
      },
      "primary_category_id": "womens-clothing-bottoms",
      "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Product-Show?pid=701644031206M",
    },
    algoliaLocalizedProduct {
      "__primary_category": {
        "0": "Femmes",
        "1": "Femmes > Vêtements",
        "2": "Femmes > Vêtements > Bas",
      },
      "categories": [
        [
          {
            "id": "newarrivals-womens",
            "name": "Femmes",
          },
        ],
        [
          {
            "id": "womens-clothing-bottoms",
            "name": "Bas",
          },
          {
            "id": "womens-clothing",
            "name": "Vêtements",
          },
          {
            "id": "womens",
            "name": "Femmes",
          },
        ],
      ],
      "color_variations": [
        {
          "color": "Combo rose vif",
          "image_groups": [
            {
              "_type": "image_group",
              "images": [
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, large",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dwcc434d54/images/large/PG.10237222.JJB52A0.PZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, large",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw58a034a4/images/large/PG.10237222.JJB52A0.BZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
              ],
              "view_type": "large",
            },
            {
              "_type": "image_group",
              "images": [
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, small",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw4e4ce4f6/images/small/PG.10237222.JJB52A0.PZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
                {
                  "_type": "image",
                  "alt": "Robe florale, Combo rose vif, small",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw2612fb5e/images/small/PG.10237222.JJB52A0.BZ.jpg",
                  "title": "Robe florale, Combo rose vif",
                },
              ],
              "view_type": "small",
            },
            {
              "_type": "image_group",
              "images": [
                {
                  "_type": "image",
                  "alt": "Robe fleurie, Mix rose vif, swatch",
                  "dis_base_link": "https://zzrk-018.sandbox.us01.dx.commercecloud.salesforce.com/on/demandware.static/-/Sites-apparel-m-catalog/default/dw506aa734/images/swatch/PG.10237222.JJB52A0.CP.jpg",
                  "title": "Robe fleurie, Mix rose vif",
                },
              ],
              "view_type": "swatch",
            },
          ],
          "variant_url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Product-Show?pid=25592581M",
        },
      ],
      "in_stock": true,
      "name": "Robe florale",
      "objectID": "701644031213M",
      "price": {
        "EUR": 92.88,
        "USD": 129,
      },
      "primary_category_id": "womens-clothing-bottoms",
      "url": "/on/demandware.store/Sites-Algolia_SFRA-Site/fr/Product-Show?pid=701644031213M",
    },
  ],
}
`;
